prime-router/src/main/kotlin/history/azure/SubmissionFunction.kt

```diff
@@ -126,4 +131,42 @@ class SubmissionFunction(
     ): HttpResponseMessage {
         return this.getDetailedView(request, id)
     }
+
+    /**
+     *Endpoint for intermediary receivers to verify status of messages. It passes
+     * a null engine to the retrieveMetadata function because Azure gets upset if there
+     * are any non-annotated parameters in the method signature other than ExecutionContext
+     * and we needed the engine to be a parameter so it can be mocked for tests
+     *
+     */
+    @FunctionName("getEtorMetadataForHistory")
+    fun getEtorMetadata(
+        @HttpTrigger(
+            name = "getEtorMetadataForHistory",
+            methods = [HttpMethod.GET],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "waters/report/{reportId}/history/etorMetadata"
+        ) request: HttpRequestMessage<String?>,
+        @BindingName("reportId") reportId: UUID,
+        context: ExecutionContext,
+    ): HttpResponseMessage {
+        return this.retrieveETORIntermediaryMetadata(request, reportId, context, null)
+    }
+
+    /**
+     * Function for finding the associated report ID that the intermediary knows about given the report ID that the sender is
+     * given from report stream
+     */
+    override fun getLookupId(reportId: UUID): UUID? {
+        val reportGraph = ReportGraph(workflowEngine.db)
+        var descendants: List<ReportFile> = emptyList()
+        workflowEngine.db.transactReturning { txn ->
```

Comment:
Why is this `getLookupId` using `transactReturning` but the one in `DeliveryFunction` isn't? `getRootReports` uses `transactReturning`, why are we doing it differently with `getDescendantReports`?

Reply:
We originally did it the same as `getRootReports` and `getRootReport`, but @mkalish [recommended](https://github.com/CDCgov/prime-reportstream/pull/14152#issuecomment-2083631465) that we do it this new way for the new `getDescendantReports` method.  We decided to not touch the pre-existing `getRootReports` or `getRootReport` methods.

Reply:
@arnejduranovic We should probably open a ticket to fix that one spot too (I think it got added recently), but the transaction getting created there makes the function unsafe to be called since you can use it while an existing transaction is in place.

Reply:
Ah, makes sense. Good catch! Will make a note to create a new ticket.